.github/scripts/delete_old_branches.py

```diff
@@ -124,6 +173,48 @@ def get_prs() -> Dict[str, Any]:
     return prs_by_branch_base
 
 
+def get_branches_with_magic_label_or_open_pr() -> Set[str]:
+    pr_infos: List[Dict[str, Any]] = []
+
+    # Grab all PRs with the magic label
+    hasNextPage = True
+    endCursor = None
+    while hasNextPage:
```

Comment:
I think you could refactor the while loop here a bit into a separate function that accepts the graphql query and a lambda function to terminate the loop, i.e. `PR_WINDOW and now - convert_gh_timestamp(pr_info["updatedAt"]) > PR_WINDOW`.  The while loops to get `GRAPHQL_ALL_PRS_BY_UPDATED_AT`, `GRAPHQL_NO_DELETE_BRANCH_LABEL`, and `GRAPHQL_OPEN_PRS` look pretty similar